oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+* If you decide that reverting will cause more harm for some reason, make sure you clearly communicate this decision.
+
+The reason it is important that only MAJOR versions include backward incompatible change is because when people upgrade dependencies, some will only read changelogs for MAJOR versions because of the potential backward incompatibilities, and may not read notes for MINOR or PATCH upgrades.
+
+Frontend Libraries
+~~~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, frontend libraries should also include a CHANGELOG.rst. However, as you'll see below, this file may simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+
+All new frontend libraries have been using an npm package called `semantic-release`_.
+
+The `semantic-release`_ package automates the following:
+
+* Incrementing the version appropriately for each PR merge, based on a well formatted commit comment. See the `semantic-release`_ README for details. A pre-commit hook is typically run to ensure the commit message is appropriately formatted for `semantic-release`_.
+
+  * One important formatting detail is to include "BREAKING CHANGE" at the beginning of a line in the commit description to get a new major version.
+
+* Publishing the changelog message as a GitHub Release in the repository, using the commit message. Here is an example `GitHub Release changelog for edx/frontend-build`_.
+
+As noted above, please update the changelogs as necessary to add clarifications or customize the messaging for the audience of the changelog, especially because these messages originally come from commit messages, which may have been written for a different audience.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may also simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Many Python libraries publish packages to PyPI when a version is tagged. Publishing a GitHub Release both tags the version and allows one to write an appropriate changelog message. This is the recommended process at this time.
+
+Every published release should have a corresponding changelog entry.
```

Comment:
Do we have to clarify what "published" means? Or is "release" enough here?

Reply:
I updated the language.  Let me know if it is not clear.